yt_dlp/extractor/yandexvideo.py

```diff
@@ -258,7 +258,7 @@ def _real_extract(self, url):
             video_id = self._match_id(redirect)
             webpage = self._download_webpage(redirect, video_id, note='Redirecting')
         data_json = self._search_json(
-            r'data\s*=', webpage, 'metadata', video_id, contains_pattern=r'{["\']_*serverState_*video.+}')
+            r'"data"\s*:', webpage, 'metadata', video_id, contains_pattern=r'{["\']_*serverState_*video.+}')
```

Comment:
```suggestion
            r'"data"\s*[=:]', webpage, 'metadata', video_id, contains_pattern=r'{["\']_*serverState_*video.+}')
```

Just in case. Same below

Reply:
Some warnings after change

yt-dlp -F https://dzen.ru/video/watch/64d21733267b6a31f46608ef
[ZenYandex] Extracting URL: https://dzen.ru/video/watch/64d21733267b6a31f46608ef
[ZenYandex] 64d21733267b6a31f46608ef: Downloading webpage
[ZenYandex] 64d21733267b6a31f46608ef: Redirecting
[ZenYandex] 64d21733267b6a31f46608ef: Downloading MPD manifest
WARNING: [ZenYandex] Ignoring subtitle tracks found in the DASH manifest; if any subtitle tracks are missing, please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
[ZenYandex] 64d21733267b6a31f46608ef: Downloading m3u8 information
WARNING: [ZenYandex] Ignoring subtitle tracks found in the HLS manifest; if any subtitle tracks are missing, please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
[info] Available formats for 64d21733267b6a31f46608ef:
ID            EXT RESOLUTION FPS │   TBR PROTO │ VCODEC        VBR ACODEC      ABR ASR MORE INFO
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────
dash-f1-a1-x3 m4a audio only     │  132k dash  │ audio only        mp4a.40.2  132k 44k [rus] DASH audio, m4a_dash

Reply:
@pukkandan Nice idea! I updated regulat expressions to capture both old and new patterns:
```
data = 
"data" :
```
The expression is `r'("data"\s*:|data\s*=)'`

Reply:
@ansdim1 Thank you! The warning exist in `master` branch as well. I guess it is the expected behavior. Anyway it is unrelated to the patch.

Reply:
The warning comes from the fact, that `self._extract_mpd_formats` and `self._extract_m3u8_formats` discard the extracted subtitles. Instead, `self._extract_mpd_formats_and_subtitles` and `self._extract_m3u8_formats_and_subtitles` should be used.
Also in the future, please don't force push. All commits will be squashed upon merge anyways.

Reply:
@seproDev I pushed another commit fixing the warning. However I can't find a video on dzen with subtitles to check it. The video posted by @ansdim1 doesn't have them. So the warning is removed, but the output is the same.

Reply:
The video posted by ansdim1 does have a sub track for me.

Reply:
I pushed another commit, fixing some tests.
@seproDev I used your suggested 'description'.
Also updated hashes for some tests.

Status of tests:
```
python3.11 test/test_download.py TestDownload.test_ZenYandex  # PASS
python3.11 test/test_download.py TestDownload.test_ZenYandex_1 # fails because of captcha
python3.11 test/test_download.py TestDownload.test_ZenYandex_2 # PASS
python3.11 test/test_download.py TestDownload.test_ZenYandex_3 # PASS

python3.11 test/test_download.py TestDownload.test_ZenYandexChannel # fails because of captcha
python3.11 test/test_download.py TestDownload.test_ZenYandexChannel_1 # fails because of captcha
python3.11 test/test_download.py TestDownload.test_ZenYandexChannel_2 # fails because of captcha
python3.11 test/test_download.py TestDownload.test_ZenYandexChannel_3 # PASS
python3.11 test/test_download.py TestDownload.test_ZenYandexChannel_4 # PASS
python3.11 test/test_download.py TestDownload.test_ZenYandexChannel_5 # fails because of captcha
```

Reply:
For me none of the tests fail due to captchas. However, these four tests still fail:
```
TestDownload.test_ZenYandexChannel   # fails due to channel no longer existing
TestDownload.test_ZenYandexChannel_1 # fails due to channel no longer existing
TestDownload.test_ZenYandexChannel_3 # fails due to channel only having 18 videos
TestDownload.test_ZenYandexChannel_4 # fails due to more than 46 videos being uploaded
```
Add `'skip': 'The page does not exist',` to the tok_media entries.
Change to `'playlist_count': 18,` for jony_me.
Change to `'playlist_mincount': 46,` for tatyanareva.

Reply:
Done. Now all `ZenYandexChannel` test cases pass for me.